pir/common/shard.go

```diff
@@ -14,3 +14,10 @@ type Shard interface {
 	GetData() []byte
 	Read(reqs []byte, reqLength int) ([]byte, error)
 }
+
+// PIRBackings is a static table of the registered / available PIR implementations
+var PIRBackings map[string]func(int, []byte, string) Shard
+
+func init() {
```

Comment:
When is the init function called? Can we just lazily create the data structure when register() is called?